### PR TITLE
Run hook on specific directories and files 

### LINKED ISSUE #50: Add default file types to run add-license-headers on
### 📝 Description of the feature

Determine which files we should have license headers on (*.py, *.cpp, *.yaml, etc.) and add "files" to the .pre-commit-hook.yaml file to make it universal

### 💡 Steps for implementing the feature

1. Determine the types of files we need headers on
2. Add "files" to the hook in .pre-commit-hook.yaml with the correct regex 

### 🔗 Useful links and references

https://[pre-commit](https://pre-commit.com/#hooks-files).com/#hooks-files
https://docs.python.org/3/library/re.html#regular-expression-syntax

### PR DESCRIPTION
**Pull request by klmcadams**
- Run the add-license-headers hook on all python files within directories named src, examples, and tests.
  - Also run the hook on all proto files. 
- Updated README with more regex examples, and how to add ``files`` to the .pre-commit-config.yaml files. 
- Included ["non_compliant"]["missing_licenses"]["MIT"] check in the function that finds files that are non-compliant.

Note: the update to check for missing_licenses requires the LICENSES/MIT.txt file to be in the root directory of the repository, otherwise it's not going to work correctly.

Closes #50

### CODE REVIEW

File: AUTHORS.md
Comment (RobPasMue) [inline AUTHORS.md:1]: Why are we renaming this file? This shouldn't be happening.

File: README.rst [hunk 1]
```diff
@@ -89,6 +89,23 @@ Currently, these hooks are available:
          in the header. Add ``--ignore_license_check`` to ignore checking for licensing information
          in the files.
 
+  #. Specify directories to run the hook on
+
+     .. note::
+
+        By default, the hook will run on proto files in any directory, as well as python files within
+        directories named src, examples, and tests. To specify additional files and/or directories
```
Comment (RobPasMue) [inline README.rst:97]:
```suggestion
        directories named ``src``, ``examples``, and ``tests``. To specify additional files and/or directories
```

File: src/ansys/pre_commit_hooks/add_license_headers.py [hunk 2]
```diff
@@ -123,9 +123,18 @@ def list_noncompliant_files(args, proj):
 
     # If ignore_license_check is False, check files for missing licensing information
     if not args.ignore_license_check:
-        missing_licensing_info = set(lint_json["non_compliant"]["missing_licensing_info"])
+        missing_licensing_info = set(
+            lint_json["non_compliant"]["missing_licensing_info"]
+            # + lint_json["non_compliant"]["missing_licenses"][args.custom_license]
```
Comment (RobPasMue) [inline src/ansys/pre_commit_hooks/add_license_headers.py:128]:
```suggestion
```
Unless strictly necessary, do not leave commented code lines

**Review by RobPasMue — APPROVED** [review]
Some minor changes - otherwise LGTM!

File: AUTHORS.md
Comment (klmcadams) [reply AUTHORS.md:1]: I thought the technical review named it Authors.md, but I just double checked and it's only AUTHORS. I'll revert the name

**Pull request merged**